### ISSUE TITLE
[python-package] Expose `is_higher_better` from metric to trained Booster

### ISSUE THREAD

**Issue by PieterPel** [microsoft/LightGBM#6945]
## Summary

Currently, when using a custom `feval` function in LightGBM, the third return value (`is_higher_better`) is used internally during training but is not preserved in the resulting `Booster` object. This makes it difficult for downstream tools (e.g., Optuna) to determine the correct optimization direction after training.

Related Optuna issue: https://github.com/optuna/optuna-integration/issues/141#issue-2419007868

## Motivation

This feature would be useful for:
- Hyperparameter tuning frameworks (like Optuna) that rely on `is_higher_better` to determine optimization direction.
- Users who want to evaluate or log model performance post-training without duplicating logic.
- Improving reproducibility and transparency in model evaluation pipelines.

## Description

In the current implementation of `train()`:

```python
booster.best_score = defaultdict(OrderedDict)
for dataset_name, eval_name, score, _ in evaluation_result_list:
    booster.best_score[dataset_name][eval_name] = score
```

Where the `is_higher_better` is not passed on to the Booster.

**Comment by jameslamb** [comment]
Thanks for using LightGBM.

> _In the current implementation of train():_

Please, instead of copying code, link to it with a commit-anchored link. That helps everyone looking at this issue to find exactly what you're referring to.

In this case, it looks like you're referring to this:

https://github.com/microsoft/LightGBM/blob/618cfafab66d19aced9878ac00c27d592a773bb4/python-package/lightgbm/engine.py#L346-L348

> _Hyperparameter tuning frameworks (like Optuna) that rely on is_higher_better to determine optimization direction._

I'm not that familiar with `optuna`. Could you share some links or a minimal, reproducible example showing how `optuna` decides when to maximize or minimize evaluation metrics?

I've reviewed https://github.com/optuna/optuna-integration/issues/141 but that has just some small code snippets, not any code links or a minimal, reproducible example.

Could you also share some pseudocode showing the interface you're proposing for LightGBM? In other words, code that doesn't work today but which you wish worked? That'd help us understand precisely what you're proposing.
